test/system/projects/samples_test.rb

```diff
@@ -25,12 +27,168 @@ class SamplesTest < ApplicationSystemTestCase
+        end
+        within('tbody') do
+          # rows
+          assert_selector '#samples-table table tbody tr', count: 3
+          # row contents
+          within("tr[id='#{@sample1.id}']") do
+            assert_selector 'th:first-child', text: @sample1.puid
+            assert_selector 'td:nth-child(2)', text: @sample1.name
+            assert_selector 'td:nth-child(3)', text: I18n.l(@sample1.created_at.localtime, format: :full_date)
+            assert_selector 'td:nth-child(4)', text: '3 hours ago'
+            assert_selector 'td:nth-child(5)', text: '2 hours ago'
+            # actions tested by role in separate test
+          end
+        end
+      end
+    end
+
+    test 'edit and delete row actions render for user with Owner role' do
+      visit namespace_project_samples_url(@namespace, @project)
+
+      within("tr[id='#{@sample1.id}'] td:last-child") do
+        assert_selector 'a', text: I18n.t('projects.samples.index.edit_button')
+        assert_selector 'a', text: I18n.t('projects.samples.index.remove_button')
+      end
+    end
+
+    test 'edit row action render for user with Maintainer role' do
+      login_as users(:joan_doe)
+      visit namespace_project_samples_url(@namespace, @project)
+
+      within("tr[id='#{@sample1.id}'] td:last-child") do
+        assert_selector 'a', text: I18n.t('projects.samples.index.edit_button')
+        assert_no_selector 'a', text: I18n.t('projects.samples.index.remove_button')
+      end
+    end
+
+    test 'no row actions for user with role < Maintainer' do
+      login_as users(:ryan_doe)
+      visit namespace_project_samples_url(@namespace, @project)
+      within('#samples-table table thead tr:first-child') do
+        # attachments_updated_at is the last column, action column does not exist
+        assert_selector 'th:last-child', text: I18n.t('samples.table_component.attachments_updated_at').upcase
+        assert_no_selector 'th:last-child', text: I18n.t('samples.table_component.action').upcase
+      end
+      within("tr[id='#{@sample1.id}'] td:last-child") do
+        assert_no_selector 'a', text: I18n.t('projects.samples.index.edit_button')
+        assert_no_selector 'a', text: I18n.t('projects.samples.index.remove_button')
+      end
+    end
+
+    test 'User with role >= Maintainer sees select and deselect buttons for samples table' do
+      visit namespace_project_samples_url(@namespace, @project)
+
+      assert_selector 'form#select-all-form'
+      assert_selector 'form#deselect-all-form'
+    end
+
+    test 'User with role < Maintainer does not see select and deselect buttons for samples table' do
+      login_as users(:ryan_doe)
+      visit namespace_project_samples_url(@namespace, @project)
+
+      assert_no_selector 'form#select-all-form'
+      assert_no_selector 'form#deselect-all-form'
+    end
+
+    test 'User with role >= Maintainer sees sample table checkboxes' do
```

Comment:
**nitpick (blocking):** Shouldn't a user with role ANALYST be able to see the checkboxes since they can launch workflow executions.

Reply:
Thanks, fixed in d60b9028eeca99d5aad54b37f1e7ce872ceb8dac